### PR TITLE
182 existing site bd fail on create

### LINKED ISSUE #182: Using `mso_schema_site_bd` with NDO 4.1+ fails with site specific configuration already created by `mso_schema_template_bd`
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.4.2

### MSO version

* V 0.9.0

### APIC version and APIC Platform for Site Level Resources

* APIC 5.2(7f)
* NDO 4.1.1k

### Affected Resource(s)

* mso_schema_site_bd

Likely other `mso_schema_site_*` resources are impacted as well.

### Terraform Configuration Files

```hcl
data "mso_tenant" "NDO1" {
  name         = "NDO1"
  display_name = "NDO1"
}

data "mso_site" "APIC1" {
  name = "APIC1"
}

resource "mso_schema" "S2" {
  name = "S2"
  template {
    name         = "T1"
    display_name = "T1"
    tenant_id    = data.mso_tenant.NDO1.id
  }
}

resource "mso_schema_site" "S2_APIC1" {
  schema_id     = mso_schema.S2.id
  site_id       = data.mso_site.APIC1.id
  template_name = "T1"
}

resource "mso_schema_template_vrf" "VRF1" {
  schema_id    = mso_schema.S2.id
  template     = "T1"
  name         = "VRF1"
  display_name = "VRF1"
}

resource "mso_schema_template_bd" "BD1" {
  schema_id     = mso_schema.S2.id
  template_name = "T1"
  name          = "BD1"
  display_name  = "BD1"
  vrf_name      = mso_schema_template_vrf.VRF1.name
  arp_flooding  = true
}

resource "mso_schema_site_bd" "BD1" {
  schema_id     = mso_schema.S2.id
  bd_name       = mso_schema_template_bd.BD1.name
  template_name = "T1"
  site_id       = data.mso_site.APIC1.id
}
```

### Debug Output

```
mso_schema_site_bd.BD1: Creating...

│ Error: "BdRef referenced on Site: APIC1 for Template  must be unique"{}
│ 
│   with mso_schema_site_bd.BD1,
│   on main.tf line 96, in resource "mso_schema_site_bd" "BD1":
│   96: resource "mso_schema_site_bd" "BD1" {
```

### Expected Behavior

The provider should update the existing site specific configuration instead of trying to create a new one.

### Actual Behavior

see above

### Steps to Reproduce

1. `terraform apply`

### PR DESCRIPTION
**Pull request by akinross**
fixes #182

### CODE REVIEW

File: mso/resource_mso_schema_site_anp.go [hunk 1]
```diff
@@ -131,14 +131,27 @@ func resourceMSOSchemaSiteAnpCreate(d *schema.ResourceData, m interface{}) error
...
+	}
 
-	_, err := msoClient.PatchbyID(fmt.Sprintf("api/v1/schemas/%s", schemaId), anpStruct)
+	if versionInt != 1 {
+		path := fmt.Sprintf("/sites/%s-%s/anps/%s", siteId, templateName, anpName)
+		anpStruct := models.NewSchemaSiteAnp("replace", path, anpRefMap)
+		_, err = msoClient.PatchbyID(fmt.Sprintf("api/v1/schemas/%s", schemaId), anpStruct)
+	}
+
+	if versionInt == 1 || err != nil {
```
Comment (sajagana) [inline mso/resource_mso_schema_site_anp.go:145]: What is the reason for `err != nil` check here?

Comment (sajagana) [inline mso/resource_mso_schema_site_anp.go:145]: What is the reason for the `err != nil` check?

Comment (akinross) [reply mso/resource_mso_schema_site_anp.go:145]: If the replace PATCH operation fails, we try will still try and add PATCH. This should never occur for ndo4 versions, but added as an additional safeguard.

**Review by sajagana — APPROVED** [review]
LGTM!

**Review by anvitha-jain — APPROVED** [review]
LGTM

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**